bazel/BUILD

```diff
@@ -105,6 +105,15 @@ config_setting(
     values = {"define": "hot_restart=disabled"},
 )
 
+# Used to avoid conflicting selects https://github.com/bazelbuild/bazel/issues/8323
```

Comment:
We might want to comment that historically we disabled hot restart implicitly for apple. And that this happened before the hot_restart define flag. It might make sense to revise if we want to continue implicitly disabling for apple, or if we want people to use the flag.